draft-normington-jsonpath-latest.txt

```diff
@@ -191,98 +191,98 @@ Internet-Draft   JavaScript Object Notation (JSON) Path   September 2020
 
    Syntactically, a JSON Path consists of a root selector ("$"), which
    selects the root node of a JSON document, followed by a possibly
-   empty sequence of _matchers_.
+   empty sequence of _selectors_.
 
-   json-path = root-selector *matcher
+   json-path = root-selector *selector
    root-selector = %x24               ; $ selects document root node
 
-   The syntax and semantics of each matcher is defined below.
+   The syntax and semantics of each selector is defined below.
 
 2.5.  Semantics
 
    The root selector "$" not only selects the root node of the input
    document, but it also produces as output a list consisting of one
    node: the input document.
 
-   A matcher may match a node and, if it matches, may then select zero
-   or more descendants of the node, or the node itself, for further
-   processing.  A syntactically valid matcher MUST NOT produce errors.
-   This means that some operations which might be considered erroneous,
-   such as indexing beyond the end of an array, simply result in fewer
-   nodes being selected.
+   A selector may select zero or more nodes for further processing.  A
+   syntactically valid selector MUST NOT produce errors.  This means
+   that some operations which might be considered erroneous, such as
+   indexing beyond the end of an array, simply result in fewer nodes
+   being selected.
 
-   But a matcher doesn't just act on a single node: each matcher acts on
-   a list of nodes and produces a list of nodes, as follows.
+   But a selector doesn't just act on a single node: each selector acts
+   on a list of nodes and produces a list of nodes, as follows.
 
    After the root selector, the remainder of the JSON Path is processed
-   by passing lists of nodes from one matcher to the next ending up with
-   a list of nodes which is the result of matching the JSON Path to the
-   input JSON document.
+   by passing lists of nodes from one selector to the next ending up
+   with a list of nodes which is the result of applying the JSON Path to
+   the input JSON document.
+
 
 
 
-Normington & Surov        Expires 27 March 2021                 [Page 4]
+Normington & Surov        Expires 28 March 2021                 [Page 4]
 
 Internet-Draft   JavaScript Object Notation (JSON) Path   September 2020
 
 
-   Each matcher acts on its input list of nodes as follows.  For each
-   node in the list, the matcher may or may not match the node.  If the
-   matcher matches the node, it selects zero or more descendants of the
-   node, or the node itself.  The output list of nodes of a matcher is
-   the concatenation of the lists of selected nodes for each input node.
+   Each selector acts on its input list of nodes as follows.  For each
+   node in the list, the selector selects zero or more nodes, each of
+   which is a descendant of the node or the node itself.  The output
+   list of nodes of a selector is the concatenation of the lists of
```

Comment:
"distinct concatenation"

Reply:
Does distinct concatenation mean with duplicates removed? That term is unfamiliar to me.

Reply:
Correct, no duplicates. I think mine currently regards the same value from different locations as distinct from one another, meaning it'll keep both.

It's worth discussing further, probably. Maybe just merge this for now, and I'll open an issue for it.

Reply:
Yes, I'd like to keep that as a separate issue. It's somewhat related to the placeholder we have in the spec about whether to remove duplicates from unions.